test/storage/block_access_controller_test.cpp

```diff
@@ -89,31 +89,31 @@ TEST(BlockAccessControllerTest, ReaderWriter) {
 // Tests that writer waits for all readers to exit
 // NOLINTNEXTLINE
 TEST(BlockAccessControllerTest, MultipleReaders) {
-  const uint32_t iteration = 100;
+  const uint32_t iteration = 10000;
   for (uint32_t i = 0; i < iteration; i++) {
     storage::BlockAccessController tested;
     tested.Initialize();
     tested.GetBlockState()->store(storage::BlockState::FROZEN);
     DECLARE_PROGRAM_POINT(reader1_acquired)
-    DECLARE_PROGRAM_POINT(reader1_released)
+    DECLARE_PROGRAM_POINT(reader1_releasing)
     DECLARE_PROGRAM_POINT(reader2_acquired)
-    DECLARE_PROGRAM_POINT(reader2_released)
+    DECLARE_PROGRAM_POINT(reader2_releasing)
     std::thread reader1([&] {
       EXPECT_TRUE(tested.TryAcquireInPlaceRead());
       PROGRAM_POINT(reader1_acquired)
+      PROGRAM_POINT(reader1_releasing)
       tested.ReleaseInPlaceRead();
-      PROGRAM_POINT(reader1_released)
     });
 
     std::thread reader2([&] {
       EXPECT_TRUE(tested.TryAcquireInPlaceRead());
       PROGRAM_POINT(reader2_acquired)
+      PROGRAM_POINT(reader2_releasing)
       tested.ReleaseInPlaceRead();
-      PROGRAM_POINT(reader2_released)
     });
     WAIT_UNTIL(reader1_acquired && reader2_acquired)
     tested.WaitUntilHot();
-    EXPECT_TRUE(REACHED(reader1_released) && REACHED(reader1_released));
+    EXPECT_TRUE(REACHED(reader1_releasing) && REACHED(reader1_releasing));
```

Comment:
I think this should be `REACHED(reader1_releasing) && REACHED(reader2_releasing)`? You have `reader1_releasing` twice here

Reply:
Good catch.